src/healthkit.cs

```diff
@@ -2142,4 +2321,62 @@ interface HKWorkoutConfiguration : NSCopying, NSSecureCoding {
 		[NullAllowed, Export ("lapLength", ArgumentSemantic.Copy)]
 		HKQuantity LapLength { get; set; }
 	}
+
+	[Watch (4, 0), iOS (11, 0)]
+	[BaseType (typeof (HKSampleType))]
+	[DisableDefaultCtor]
+	interface HKSeriesType {
+		[Static]
+		[Export ("workoutRouteType")]
+		HKSeriesType WorkoutRouteType { get; }
+	}
+
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface HKSeriesBuilder {
+		[Export ("discard")]
+		void Discard ();
+	}
+
+	[BaseType (typeof(HKSample))]
+	[DisableDefaultCtor]
+	interface HKSeriesSample : NSCopying {
+		[Export ("count")]
+		nuint Count { get; }
+	}
+
+	[BaseType (typeof(HKSeriesSample))]
+	[DisableDefaultCtor]
+	interface HKWorkoutRoute : NSCopying {
+		[Watch (4, 0), iOS (11, 0)]
+		[Field ("HKWorkoutRouteTypeIdentifier")]
+		NSString TypeIdentifier { get; }
+	}
+
+	[Watch (4, 0), iOS (11, 0)]
+	[BaseType (typeof (HKSeriesBuilder))]
+	[DisableDefaultCtor]
+	interface HKWorkoutRouteBuilder {
+		[Export ("initWithHealthStore:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (HKHealthStore healthStore, [NullAllowed] HKDevice device);
+
+		[Export ("insertRouteData:completion:")]
+		void InsertRouteData (CLLocation [] routeData, Action<bool, NSError> completion);
+
+		[Export ("finishRouteWithWorkout:metadata:completion:")]
+		void FinishRouteWithWorkout (HKWorkout workout, [NullAllowed] NSDictionary<NSString, NSObject> metadata, Action<HKWorkoutRoute, NSError> completion);
```

Comment:
I wonder if you should make this `[Protected]` and be named `FinishRoute`

Reply:
Why Protected?  The name change should be fine?

Reply:
Since we offer a better option which is the one that takes a `HKMetadata metadata` I think we can expose just this one and "hide" the one that takes a `NSDictionary` and only make it available if this object gets subclassed.

OTOH we can keep both as overloads named `FinishRoute` and let the user pick the one he wants also I would turn  `NSDictionary<NSString, NSObject>` to just a `NSDictionary` and remove the generic part since we already provide a better overload.

Reply:
Also I think you will have some build issues here 

```
[Async, Wrap ("FinishRoute (workout, metadata, completion)")]
```

Because the weakly typed method is called `FinishRouteWithWorkout` 😉

Reply:
yep (recursion), `FinishRoute` sounds better too